jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -352,7 +352,10 @@ public PushBuilder newPushBuilder()
             HttpHeader header = field.getHeader();
             if (header == HttpHeader.SET_COOKIE)
             {
-                HttpCookie cookie = ((SetCookieHttpField)field).getHttpCookie();
+                HttpCookie cookie = (field instanceof SetCookieHttpField)
+                    ? ((SetCookieHttpField)field).getHttpCookie()
+                    : new HttpCookie(field.getValue());
```

Comment:
This reparse should follow the `HttpConfiguration` setting for `response._responseCookieCompliance` too.

Reply:
Hmm maybe.  But that is about the format of the set-cookies that jetty will generate.  This is a set-cookie that the application has generated, probably to get around some compliance issue.
Also, we don't actually have a compliance specific Set-Cookie parser, as we currently only use that part for generation.  So that would be a bit of work.    

So I think this is a step in the right direction, if not perfect.

Reply:
The normal cookie parsing step is done via the `CookieCutter` class, which has compliance modes, not the direct `org.eclipse.jetty.http.HttpCookie(String)` constructor, which uses the `java.net.HttpCookie` class to parse, which has limitations (namely it only supports RFC2109 and RFC2965).

Reply:
CookieCutter only parses Cookie values, not Set-Cookie values... and then it is mostly about ancient long deprecated RFCs.

I've extended the test to make sure we don't throw anything if the same-site attribute is set.  We don't.  It is ignored, but then all we need is max-age to be parsed correctly, which it is.
So I still think we are good.   We don't need to parse same-site, as by definition the cookie on a push request is going to the same site.

Reply:
ok, works for me.